_posts/2017/2017-04-19-build-a-multiple-repository-pull-request-test.md

```diff
@@ -0,0 +1,161 @@
+Developing effort and the risk of loss of quality increases significantly, but you can solve the situation with building up a multiple repositories pull request test.
+
+### 1. How do we prepare the test?
+The solution is as simple as complicated: Your Pull request has to know, which code change belongs to a specific feature.
+While using three different repositories, how can you achieve to test all changes at once?
+Three things are necessary:
+* GitHub Organizational Plugin,
+* Pipeline plugin,
+* naming convention.
+
+### 2. Why do we need those things?
+You can use the [Github Organizational Folder Plugin](https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Organization+Folder+Plugin) to check your repositories for new Pull request.
+This plugin will check every affected repository for changes and activate the Jenkinsfile in the main directory, if a change was made.
+
+You can create a [Jenkinsfile](https://jenkins.io/doc/book/pipeline/jenkinsfile/) at a repository, which will be triggered after a Pull request.
+The task of this Jenkinsfile is mainly to check, if branches with the same name from the same developer in other repositories exists and will use them for the PR tests, if a change was detected.
+
+You have to agree to naming conventions, mean you have to set the same name for all affected branches in the different repositories.
+As you work with Pull requests, you need to find out the branch name first.
+Luckily, you can get the pull request number directly from the trigger.
+You need to [GET the PR information](https://developer.github.com/v3/pulls/#get-a-single-pull-request) from Github:
+````
+BRANCH = git.getBranchNameFromPr("rest-test-gold", env.CHANGE_ID)
+````
+````
+def String getBranchNameFromPr(String repository, String changeID) {
+    def getBranchNameURL = 'https://api.github.com/repos/ePages-de/' + repository + '/pulls/' + changeID
+    return this.getFromGithubAPI(getBranchNameURL).head.ref
+}
+````
+ ````
+    ...
+    // Evaluate, if branches exist in the given repository.
+    if (git.getBranchForRepository("rest-test-gold", BRANCH, REMOTE)) {
+        println "We can use " + BRANCH + " for rest-test-gold..."
+        RTG_BRANCH_NAME = BRANCH
+    } else {
+        // Set the branch and remote name to standard, if branch do not exist.
+        println "We can use dev for rest-test-gold..."
+        RTG_BRANCH_NAME = "dev"
+    }
+    ...
+ ````
+As you now have the name of the branch, you to do the same for the remote name of the developer, who created the pull request.
+
+How can we now ensure, that the PR is only triggered once and not for every repository, where a Pull request was set from a developer?
+
+The solution is to build a hierarchy into the repository structure.
+As every repository has a describing [Jenkinsfile](https://jenkins.io/doc/book/pipeline/jenkinsfile/), we need a kind of control repository.
+
+{% image blog/blog-pr-test-1.png %}
```

Comment:
the pictures are too large, see attached screenshot. please resize.
![snippet-image](https://cloud.githubusercontent.com/assets/10460958/24037439/1ac5e408-0afe-11e7-8b68-e34fd50ad52a.png) (here you can also see, that the code snippet is not rendered correctly.)